pkgs/development/tools/rust/itm/default.nix

```diff
@@ -0,0 +1,27 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, rustPlatform
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "itm";
+  version = "0.3.1";
+
+  src = fetchFromGitHub {
+    owner = "rust-embedded";
+    repo = pname;
+    rev = "3e4251b42aa2e4b05ae372c47c7b835b8acae6dc";
+    sha256 = "15pa0ydm19vz8p3wairpx3vqzc55rp4lgki143ybgw44sgf8hraj";
+  };
+
+  cargoSha256 = "1d748vh1ig8r0cd70lsnw9fdmanic4jxxibbbagd5mrh8462v64v";
+  cargoPatches = [ ./itm_cargo_lock.patch ];
```

Comment:
I'm curious why this is the necessary. Please upstream if it fixes the build.

Reply:
Without patching the Cargo.lock file I get build errors:

```
➜ nix-build -A itm
these derivations will be built:
  /nix/store/b95c43zmcs2kz1mwy145k7fzpqpy97yy-itm-0.3.1.drv
building '/nix/store/b95c43zmcs2kz1mwy145k7fzpqpy97yy-itm-0.3.1.drv'...
unpacking sources
unpacking source archive /nix/store/zz34rw0bgrwg840hmwd9c3mai01ivv5v-source
source root is source
Executing cargoSetupPostUnpackHook
unpacking source archive /nix/store/dm1rj9v3bz623726f3w10y93kg0521jx-itm-0.3.1-vendor.tar.gz
Finished cargoSetupPostUnpackHook
patching sources
Executing cargoSetupPostPatchHook
Validating consistency between /build/source//Cargo.lock and /build/itm-0.3.1-vendor.tar.gz/Cargo.lock
Finished cargoSetupPostPatchHook
configuring
building
Executing cargoBuildHook
++ env CC_x86_64-unknown-linux-gnu=/nix/store/s5hkav7whndbfz0szshpb46h4idqdq9a-gcc-wrapper-10.3.0/bin/cc CXX_x86_64-unknown-linux-gnu=/nix/store/s5hkav7whndbfz0szshpb46h4idqdq9a-gcc-wrapper-10.3.0/bin/c++ CC_x86_64-unknown-linux-gnu=/nix/store/s5hkav7whndbfz0szshpb46h4idqdq9a-gcc-wrapper-10.3.0/bin/cc CXX_x86_64-unknown-linux-gnu=/nix/store/s5hkav7whndbfz0szshpb46h4idqdq9a-gcc-wrapper-10.3.0/bin/c++ cargo build -j 4 --target x86_64-unknown-linux-gnu --frozen --release
error: the lock file /build/source/Cargo.lock needs to be updated but --frozen was passed to prevent this
If you want to try to generate the lock file without accessing the network, remove the --frozen flag and use --offline instead.
builder for '/nix/store/b95c43zmcs2kz1mwy145k7fzpqpy97yy-itm-0.3.1.drv' failed with exit code 101
error: build of '/nix/store/b95c43zmcs2kz1mwy145k7fzpqpy97yy-itm-0.3.1.drv' failed
```

I remember seeing prior art around patching the `Cargo.lock` file like this which is why I've gone this route. I think this happens when the `Cargo.lock` file is out of date when a release commit is tagged and figured I'd do the same here. I'll try and communicate the issue to the maintainers in the meantime.